odoo/orm/registry.py

```diff
@@ -654,6 +657,21 @@ def init_models(self, cr: Cursor, model_names: Iterable[str], context: dict[str,
             del self._foreign_keys
             del self._is_install
 
+    def check_null_constraints(self, cr: Cursor) -> None:
+        """ Check that all not-null constraints are set. """
+        cr.execute("SELECT table_name, column_name FROM information_schema.columns WHERE is_nullable = 'NO' AND table_schema = 'public'")
+        not_null_columns = set(cr.fetchall())
+
+        self.not_null_fields.clear()
+        for Model in self.models.values():
+            if Model._auto and not Model._abstract:
+                for field in Model._fields.values():
+                    if field.column_type and field.store and field.required:
+                        if (Model._table, field.name) in not_null_columns:
+                            self.not_null_fields.add(field)
```

Comment:
Here we use the field onbject in the `self.not_null_fields`
When manual fields are modifed in `ir.model.field`, `_setup_models__` and `init_models` will be called but not this `check_null_constraints`, so I guess the new field object maynot be the old one?

Reply:
Don't we create a new registry in that case?

Reply:
No, the registry is [invalidated](https://github.com/odoo/odoo/blob/8263ddc408fb9bef646558b8ae91a9de3b3b6cc8/odoo/orm/registry.py#L371) in `_setup_models__`. And the sequence of the current worker is [increased](https://github.com/odoo/odoo/blob/8263ddc408fb9bef646558b8ae91a9de3b3b6cc8/odoo/orm/registry.py#L954) at the end of the transaction to avoid reloading the registry.

I think it is base on the logic that manual fields are always toplevel which won't be overridden. So we don't need to reload the whole registry to get correct inheritence.